front-end/src/components/gallery/Gallery.js

```diff
@@ -108,13 +179,11 @@ const Gallery = () => {
     }
 
     const addFilterOptions = (filterType, value) => {
-        const filter = {};
         if (filterType === 'category') {
             if (value) {
-                filter[filterType] = value;
-                setCategoryOption(filter)
+                setCategoryOption(value)
             } else {
-                setCategoryOption({})
+                setCategoryOption('')
             }
         } else if (filterType === 'size') {
```

Comment:
Why don't you extract the names of the filters in a constant variable so you can reuse it here and both in the useEffect. This way you will avoid lexical mistakes.